dwa_local_planner/src/dwa_planner.cpp

```diff
@@ -205,8 +205,18 @@ namespace dwa_local_planner {
     return true;
   }
 
+  inline bool is2DPoseEqual(const geometry_msgs::PoseStamped& a, const geometry_msgs::PoseStamped& b) {
+    return a.pose.position.x == b.pose.position.x && a.pose.position.y == b.pose.position.y &&
+        a.pose.orientation.z * (a.pose.orientation.w >= 0 ? 1 : -1) ==
```

Comment:
Might be better to normalise the angle and check if it is near zero.

Reply:
For better understandability or do you see an actual issue with this code? It's just less complex of an operation compared to `normalize_angle(2 * atan2(q1.z, q1.w) - 2 * atan2(q2.z, q2.w)) == 0`.

What about `a.pose.orientation.z * b.pose.orientation.w == a.pose.orientation.w * b.pose.orientation.z`?

Reply:
for better understanding